src/java/org/apache/cassandra/db/CounterMutation.java

```diff
@@ -175,18 +201,20 @@ private void grabCounterLocks(Keyspace keyspace, List<Lock> locks) throws WriteT
             try
             {
                 if (!lock.tryLock(timeout, NANOSECONDS))
-                    handleLockTimeout(replicationStrategy);
+                    handleLockTimeout(replicationStrategy, startTime);
                 locks.add(lock);
             }
             catch (InterruptedException e)
             {
-                handleLockTimeout(replicationStrategy);
+                handleLockTimeout(replicationStrategy, startTime);
             }
         }
+        lockAcquireTime.addNano(System.nanoTime() - startTime);
     }
 
-    private void handleLockTimeout(AbstractReplicationStrategy replicationStrategy)
+    private void handleLockTimeout(AbstractReplicationStrategy replicationStrategy, long startTime)
```

Comment:
let's use the occasion to rename it to `handleLockTimeoutAndThrow` so that the body of `try` above doesn't look that weird